### PR TITLE
Improve the coding style and fix issue related to acc parallel directive

### PR DESCRIPTION
**Pull request by sjsprecious**
This PR mainly:
- clean the GPU code in `domaindiag.F`
- fix some `acc parallel` directives that may lead to wrong results without setting module variables correctly on the device

Verification test:

CPU (intel) versus OpenACC (nvhpc)
29 stat variables are identical.
48 stat variables show a mean relative difference > 1e-06
6 stat variables show a mean relative difference <= 1e-06
four variables with largest error: KSVMAX, KSHMAX, TMW, TKEMAX

CPU (intel) versus MPI+OpenACC (nvhpc)
26 stat variables are identical.
57 stat variables show a mean relative difference > 1e-06
0 stat variables show a mean relative difference <= 1e-06
four variables with largest error: STHPMX, THPMAX, TMW, PPMAX

### CODE REVIEW

File: src/domaindiag.F [hunk 1]
```diff
@@ -2291,7 +2324,7 @@ subroutine domaindiag(nstep,mtime,nwritet,trecs,trecw,qname,qunit,dt,dosfcflx, &
           ! cape, cin, lfc, lcl:
 
           !$omp parallel do default(shared)  &
-          !$omp private(i,j,k,pfoo,tfoo,qfoo,zel,psource,tsource,qvsource)
+          !$omp
```
Comment (johnmauff) [inline src/domaindiag.F]: @sjsprecious, why are you modifying $omp directives here.  I thought that the presence of private() declarations are necessary to ensure correctness.

File: src/domaindiag.F [hunk 2]
```diff
@@ -2364,19 +2397,17 @@ subroutine domaindiag(nstep,mtime,nwritet,trecs,trecw,qname,qunit,dt,dosfcflx, &
...
 !END-BLOCK02
 #endif
-     !stop 'domaindiag: after BLOCK02'
 
       !c-c-c-c-c-c-c-c-c-c
 
       IF( imoist.eq.1 .and. th0(1,1,nk)*pi0(1,1,nk).lt.to )THEN
 
-        !$omp parallel do default(shared) private(i,j,k,tx,tlast)
+        !$omp parallel do default(shared)
```
Comment (johnmauff) [inline src/domaindiag.F]: @sjsprecious  We should not be modifying $omp directives please revert this.

File: src/domaindiag.F [hunk 3]
```diff
@@ -1573,71 +1602,73 @@ subroutine domaindiag(nstep,mtime,nwritet,trecs,trecw,qname,qunit,dt,dosfcflx, &
...
+      !$acc parallel loop gang vector default(present)
       do k=1,nk
         ! use minimum of 1.0e-5 and 1% of saturation wrt liquid (using last known t,p)
         qcrit(k) = min( 1.0e-5 , 0.01*rslf(pavg(k),thavg(k)*((pavg(k)*rp00)**rovcp)) )
       enddo
+      !$acc end parallel
 
-        ! use minimum of 1.0e-5 and 1% of saturation wrt liquid (using last known t,p)
-        !qcrit(k) = min( 1.0e-5 , 0.01*rslf(pavg(k),thavg(k)*((pavg(k)*rp00)**rovcp)) )
-        if( nqc.ge.1 )then
```
Comment (johnmauff) [inline src/domaindiag.F:1585]: @sjsprecious this looks like a pretty large number of changes.  Is there any way that we can break them into separate PRs?

Comment (johnmauff) [reply src/domaindiag.F:1585]: After review these changes look good.

File: src/domaindiag.F (hunk 2 above)
Comment (johnmauff) [reply src/domaindiag.F]: Thanks for reverting the omp directives

File: src/domaindiag.F (hunk 1 above)
Comment (johnmauff) [reply src/domaindiag.F]: Thanks for reverting the omp directive changes

**Review by johnmauff — APPROVED** [review]
Thanks, @sjsprecious this looks good now.

**Review by sherimickelson — APPROVED** [review]
Thanks @sjsprecious .  Everything looks good.
